### PR TITLE
17186 - UXA for alterations + Fixes

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*[ /bcgov/entity17186](https://github.com/bcgov/entity/issues/17186)

*Description of changes:*
overall UX check for alteration flow - changing business types

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/YourCompany/ChangeBusinessType.vue [hunk 1]
```diff
@@ -522,6 +536,15 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
   }
 }
 
+/*
+This rule changes the color and opacity of the text in a disabled v-select component.
+The :deep pseudo-element and !important flag are used to override Vuetify's default styles.
+*/
+:deep(.disabled-select.theme--light.v-select .v-select__selection--disabled) {
```
Comment (Jxio) [inline src/components/common/YourCompany/ChangeBusinessType.vue:543]: this one is for:
image

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue [hunk 2]
```diff
@@ -235,22 +235,25 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
...
     const entityTypeOptions = this.getResource?.changeData?.entityTypeOptions
     const isValidConversionNameRequest = nr.request_action_cd === NrRequestActionCodes.CONVERSION &&
-          entityTypeOptions?.some(options => options.value === nr.legalType)
+      entityTypeOptions?.some(options => options.value === nr.legalType)
     return (isNameEntityTypeDifferent && !isValidConversionNameRequest)
   }
 
   /* Generate content of error depending on name request type. */
   nameRequestErrorText (nr: NrResponseIF): string {
+    let changeFrom = GetCorpFullDescription(nr.actions[0].filingName.split(' ')[0] as CorpTypeCd)
```
Comment (Jxio) [inline src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue:244]: It was nr.entity_type_cd here before, but it's not entity "changed from", actions is the only attribute contains changed from entity type, here is an example:
image

Comment (seeker25) [reply src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue:244]: I don't think we should do this, this is fairly hacky, we should load in the correct entity_type_cd, so it populates the display. I can hunt down the data if needed.

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue [hunk 3]
```diff
@@ -235,22 +235,25 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
...
     const isConversionOrAlterationNameRequest = nr.request_action_cd === NrRequestActionCodes.CONVERSION
     let dialogContent = ''
     if (isConversionOrAlterationNameRequest) {
-      dialogContent = `<p class="info-text">This alteration name request from
-        ${GetCorpFullDescription(nr.entity_type_cd)} to ${GetCorpFullDescription(nr.legalType)}
+      dialogContent = `<p class="info-text">
+        This alteration name request from
```
Comment (seeker25) [inline src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue:249]: Revert this please, we can get the enums from Names and put them in

File: src/store/store.ts [hunk 4]
```diff
@@ -150,6 +150,28 @@ export const useStore = defineStore('store', {
       return (this.getCorrectedFilingType === FilingTypes.CHANGE_OF_REGISTRATION)
     },
 
+    /** The origin entity type. */
+    getOriginLegalType (): CorpTypeCd {
```
Comment (seeker25) [inline src/store/store.ts:154]: This is good =)

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue (hunk 2 above)
Comment (Jxio) [reply src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue]: UX updated

**Review by seeker25 — APPROVED** [review]
Nice job

File: src/components/common/YourCompany/ChangeBusinessType.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/common/YourCompany/ChangeBusinessType.vue:543]: Awesome comment! Now I know exactly what/why this is doing. Thanks!

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue [hunk 5]
```diff
@@ -244,13 +244,13 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
...
     let dialogContent = ''
     if (isConversionOrAlterationNameRequest) {
-      dialogContent = `<p class="info-text">This alteration name request from
-        ${GetCorpFullDescription(nr.entity_type_cd)} to ${GetCorpFullDescription(nr.legalType)}
-        does not match the current business type
+      dialogContent = `<p class="info-text">
+        This alteration name request does not match the current business type
         <b>${GetCorpFullDescription(this.getEntityType)}</b>.\n\n
-        The Name Request type must match the business type before you can continue.</p>`
+        The Name Request type must match the busines type before you can continue.</p>`
```
Comment (severinbeauvais) [inline src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue:250]: New typo!

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue [hunk 6]
```diff
@@ -244,13 +244,13 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
...
     } else {
-      dialogContent = `<p class="info-text">This ${GetCorpFullDescription(nr.entity_type_cd)}
+      dialogContent = `<p class="info-text">
+      This ${GetCorpFullDescription(nr.legalType)}
       Name Request does not match the current business type
       <b>${GetCorpFullDescription(this.getEntityType)}</b>.\n\n
       The Name Request type must match the business type before you can continue.</p>`
```
Comment (severinbeauvais) [inline src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue:256]: Funny that lint didn't complain that lines 253-256 should be indented...? I think they should. (Let me know if lint says they shouldn't.)

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue]: I agree with Travis: look at the Request Type Code.

See also: https://github.com/bcgov/bcrs-shared-components/blob/main/src/enums/name-request-codes.ts

**Review by severinbeauvais — APPROVED** [review]
Just a couple of tiny issues but LGTM. Nicely done 👍

**Review by JazzarKarim — APPROVED** [review]
Looks great Jia! 👍

File: src/store/store.ts [hunk 7]
```diff
@@ -150,6 +150,28 @@ export const useStore = defineStore('store', {
       return (this.getCorrectedFilingType === FilingTypes.CHANGE_OF_REGISTRATION)
     },
 
+    /** The origin entity type. */
```
[the remark this answers is not in the document]
Comment (Jxio) [reply src/store/store.ts]: fixed

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue (hunk 6 above)
Comment (Jxio) [reply src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue]: just fixed indention. and re-run Lint before fix, it doesn't complain any.

**Comment by Jxio** [comment]
Thanks for all of your review!

**Comment by severinbeauvais** [comment]
Are you ready to merge?

**Comment by Jxio** [comment]
> Are you ready to merge?

yes it's ready to merge

**Pull request merged**